### PR TITLE
Vqm2bliff one lut removal

### PR DESCRIPTION
**Pull request by helen-23**
Running Quartus Fitter using a Stratix IV device can introduce some one-LUTs in the post-fit netlist that makes it harder for VPR to place and route. So if we want to run VPR with the post-fit netlist, the existence of these one-LUTs can hurt wire length and timing results. Adding a function in vqm2bliff to get rid of these one-LUTs before converting the post-fit VQM to BLIF

#### Description

In the cleanup netlist stage of VQM2BLIF, a netlist is built and the buffer/inverter nets are removed. I've added a function in this stage that serves to remove any feeder one-LUTs in the netlist as well.

#### Motivation and Context

VPR performs better with the one-LUTs removed.

#### How Has This Been Tested?

Tested on some large Titan benchmarks such as the FFT and DLA variants

#### Types of changes

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: utils/vqm2blif/src/base/cleanup.cpp [hunk 1]
```diff
@@ -223,6 +238,154 @@ void add_subckts (t_node** nodes, int num_nodes, busvec* buses, struct s_hash**
...
+        one LUTs from the netlist, if they exist, before converting it into the BLIF file format.
+
+	Go through all nodes, if a node's source net is driven by a one-LUT-type node and if this source net only has one
+        child (the node itself):
+
+	  1. If the one-LUT has an input and an output (signal passthrough):
+             Re-associate the node's input port with the source net of the one-LUT, then remove the one-LUT and the node's original
+             source net.
+
+             -----                -------                 -----
```
[the remark this answers is not in the document]
Comment (helen-23) [reply utils/vqm2blif/src/base/cleanup.cpp:256]: Haha believe it or not I had a ton of fun making it too. Glad it helps!

File: utils/vqm2blif/src/base/cleanup.cpp [hunk 2]
```diff
@@ -627,5 +790,64 @@ void print_all_nets ( busvec* buses, const char* filename ){
 
 	outfile.close();
 }
+
+//============================================================================================
+//============================================================================================
+
+bool is_onelut ( t_node* node ) {
```
[the remark this answers is not in the document]
Comment (helen-23) [reply utils/vqm2blif/src/base/cleanup.cpp]: This is a really good point. I agree.

**Comment by helen-23** [comment]
Hi Vaughn,
Thanks for reviewing! I've updated all changes as suggested.

**Comment by vaughnbetz** [comment]
Looks good; will merge as soon as CI is finished.

**Comment by helen-23** [comment]
Hi Vaughn,
I believe this one is also ready for merging, unless I've accidentally missed something. It's saying that there is one change requested but I can't find where it is. Maybe that's just the old request you had earlier? Thanks!

**Pull request merged**